src/VisualStudio/CSharp/Impl/Snippets/SnippetCommandHandler.cs

```diff
@@ -73,11 +87,33 @@ public CommandState GetCommandState(SurroundWithCommandArgs args)
             return CommandState.Available;
         }
 
+        public CommandState GetCommandState(TypeCharCommandArgs args, Func<CommandState> nextCommandHandler)
+        {
+            return nextCommandHandler();
+        }
+
+        public void ExecuteCommand(TypeCharCommandArgs args, Action nextCommandHandler, CommandExecutionContext executionContext)
+        {
+            AssertIsForeground();
+            if (args.TypedChar == ';'
+                && AreSnippetsEnabled(args)
+                && args.TextView.Properties.TryGetProperty(typeof(AbstractSnippetExpansionClient), out AbstractSnippetExpansionClient snippetExpansionClient)
+                && snippetExpansionClient.IsFullMethodCallSnippet)
+            {
+                // Commit the snippet. Leave the caret in place, but clear the selection. Subsequent handlers in the
+                // chain will handle the remaining Complete Statement (';' insertion) operations.
+                snippetExpansionClient.CommitSnippet(leaveCaret: true);
+                args.TextView.Selection.Clear();
```

Comment:
clearing the selection seems a bit odd.  could you clarify what scenario that is for?  does committing the snippet not normally do that?  would it make more sense to put in the client code? (or do we not own htat)?

Reply:
Committing the snippet either leaves the caret as-is, or moves it to `$end$`. Leaving it in place does not clear the selection.

Reply:
I'm not sure what I can do here since the behavior is already documented.

Reply:
i agree with jason on this.  feels cleaner if we can move selection clearing to auto-ender handler.

Reply:
> ... does it not move and clear the selection implicitly?

Complete Statement takes no action if there is an active selection.

Reply:
> Complete Statement takes no action if there is an active selection

Alright if nothing else that's the comment that's missing here to document the "why?" since that wasn't obvious.

Reply:
⏱ Rewriting as follows:

> Commit the snippet. Leave the caret in place, but clear the selection. Subsequent handlers in the chain will handle the remaining Complete Statement (';' insertion) operations **only if there is no active selection**.

Reply:
Works for me, @CyrusNajmabadi?